stemcell_builder/stages/system_azure_init/assets/etc/waagent/waagent.conf

```diff
@@ -85,8 +85,8 @@ OS.OpensslPath=None
 # Set the path to SSH keys and configuration files
 OS.SshDir=/etc/ssh
 
-# Set /etc/ssh/sshd_config ClientAliveInterval to 900
-OS.SshClientAliveInterval=900 # NON-DEFAULT, works around azure TCP NAT tables timeouts
```

Comment:
Are we concerned about the comment here about `TCP NAT tables timeouts` ?

Reply:
This is a good question. I *think* we will be okay, but it is not 100% clear. It is difficult to find context on this "TCP NAT table timeout" issue referenced in the comment.

The interval was originally set in `waagent.conf` as part of https://github.com/cloudfoundry/bosh-linux-stemcell-builder/pull/88, which removed ClientAliveInterval from the `sshd_config`. However, it was added back to the `sshd_config` soon after in https://github.com/cloudfoundry/bosh-linux-stemcell-builder/commit/5d8f198eb23d1a83e010e8f4dbc06d9428b5619b and later was lowered only in the `sshd_config` as part of https://github.com/cloudfoundry/bosh-linux-stemcell-builder/pull/111

I'm unsure of how the settings in `waagent.conf` get applied, but can definitely confirm that the contents of `sshd_config` on a BOSH-deployed Azure VM match those generated by `base_ssh/apply.sh`, and don't seem to be overwritten by the waagent settings. Which implies we've been running with an interval of 300 second since March 2020 with no reports of problems. Since we don't know what the original problem was, it's unclear if the problem is just gone now or if 180 will be "too low" and cause the problem to recur.

```
test-bu-1/83f80230-0db9-40ab-b588-d14161ea97b3:~# cat /etc/ssh/sshd_config

# This is the sshd server system-wide configuration file.  See
# sshd_config(5) for more information.

... snip ...

IgnoreRhosts yes
ClientAliveInterval 300
LoginGraceTime 60

... snip ...
```